packages/playwright-core/src/server/browserContext.ts

```diff
@@ -665,10 +665,7 @@ export async function createClientCertificatesProxyIfNeeded(options: channels.Br
   if ((options.proxy?.server && options.proxy?.server !== 'per-context') || (browserOptions?.proxy?.server && browserOptions?.proxy?.server !== 'http://per-context'))
     throw new Error('Cannot specify both proxy and clientCertificates');
   verifyClientCertificates(options.clientCertificates);
-  const clientCertificatesProxy = new ClientCertificatesProxy(options);
-  options.proxy = { server: await clientCertificatesProxy.listen() };
-  options.ignoreHTTPSErrors = true;
-  return clientCertificatesProxy;
+  return new ClientCertificatesProxy(options);
```

Comment:
- This method should probably take `types.BrowserContextOptions`. I also see 49 occurrences of `BrowserNewContext` in `src/server` that should probably all be replaced.
- Once we remove line 665, we can move the validation logic into `ClientCertificatesProxy` constructor and remove this method entirely.